web/react/components/sidebar.jsx

```diff
@@ -197,6 +197,14 @@ module.exports = React.createClass({
             if (UserStore.getCurrentId() === msg.user_id) {
                 AsyncClient.getChannels(true);
             }
+        } else if(msg.action === "user_removed") {
+            if(msg.user_id === UserStore.getCurrentId()) {
+                AsyncClient.getChannels(true);
+
+                if(msg.props.channel_id === ChannelStore.getCurrentId()) {
+                    window.location.reload();
```

Comment:
The timeout doesn't work either, it was another method I tried. However, if we're going for the modal you suggested below, then we can fire it on close and that seems to negate whatever effect was occurring.